pkg/platform/kube/logProxy/elastic.go

```diff
@@ -60,10 +66,222 @@ func NewElasticLogProxy(config *platformconfig.ElasticSearchConfig) (*ElasticLog
 	return esClient, err
 }
 
+func (e *ElasticLogProxy) GetFunctionReplicas(ctx context.Context, options *GetFunctionReplicaOptions) ([]string, error) {
+	if e.client == nil {
+		return nil, errors.New("Elasticsearch client is not configured")
+	}
+
+	searchRequest := e.getFunctionBaseSearchRequest(options.FunctionName)
+
+	// Set to 0 to avoid fetching actual log documents (hits).
+	// We're only interested in aggregation results (e.g., distinct pod names),
+	// so setting this to 0 reduces response size, improves query performance,
+	// and saves bandwidth and memory.
+	searchRequest.Size = common.Pointer(0)
+
+	if err := e.addTimeFilter(searchRequest, options.TimeFilter); err != nil {
+		return nil, errors.Wrap(err, "Failed to apply time filter")
+	}
+
+	// aggregate by pod names
+	searchRequest.Aggregations = map[string]types.Aggregations{
+		"distinct_pod_names": {
+			Terms: &types.TermsAggregation{
+				Field: common.Pointer("kubernetes.pod.name"),
+				Size:  common.Pointer(100000),
+			},
+		},
+	}
+
+	resp, err := e.client.Search().
+		Index(e.index).
+		Request(searchRequest).
+		Do(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to execute search")
+	}
+
+	agg, ok := resp.Aggregations["distinct_pod_names"]
+	if !ok {
+		return nil, errors.New("Failed to get 'distinct_pod_names' in response from ES")
+	}
+
+	// Convert to typed aggregation:
+	termsAgg, ok := agg.(*types.StringTermsAggregate)
+	if !ok {
+		return nil, errors.New("Failed to convert aggregations to StringTermsAggregate")
+	}
+
+	replicas := make([]string, 0)
+
+	switch buckets := termsAgg.Buckets.(type) {
+	case []types.StringTermsBucket:
+		for _, bucket := range buckets {
+			replicas = append(replicas, bucket.Key.(string))
+		}
+	case map[string]types.StringTermsBucket:
+		for _, bucket := range buckets {
+			replicas = append(replicas, bucket.Key.(string))
+		}
```

Comment:
1. How come it can be different types?
2. If we anyway do the same `for` loop for both type - do we need the switch-case for the Go compiler or can we skip it?

Reply:
@TomerShor this is specifics of es go client, it can be both:

```
// BucketsStringTermsBucket holds the union for the following types:
//
//	map[string]StringTermsBucket
//	[]StringTermsBucket
//
```
depending on how they serialised. I added support for both to stay on the safe side.